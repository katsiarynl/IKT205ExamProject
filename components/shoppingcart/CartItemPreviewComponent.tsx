import { View, Text } from "react-native";
import React from "react";
import QuantityComponent from "./QuantityComponent";
export default function CartItemPreviewComponent(props) {
  return (
    <View
      style={{
        backgroundColor: "transparent",
        flex: 1,
        flexDirection: "column",
      }}
    >
      <View
        style={{
          backgroundColor: "transparent",
          flexDirection: "row",
          flex: 1,
        }}
      >
        <View style={{ backgroundColor: "transparent", flex: 4 }}>
          <Text style={{ fontSize: 30 }}> {props.name}</Text>
        </View>
        <View
          style={{
            backgroundColor: "transparent",
            flex: 2,
            alignItems: "center",
            justifyContent: "center",
          }}
        >
          <Text style={{ fontSize: 20, fontWeight: "bold" }}>
            {props.price}
          </Text>
        </View>
      </View>
      <View style={{ flex: 3, backgroundColor: "#F4F4F8" }}>
        <Text style={{ fontSize: 18 }}>{props.description} </Text>
      </View>

      <QuantityComponent quantity={props.quantity} id={props.id} />
    </View>
  );
}
