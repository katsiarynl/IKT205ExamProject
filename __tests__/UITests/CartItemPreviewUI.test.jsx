import CartItemPreviewComponent from "../../components/shoppingcart/CartItemPreviewComponent";
import { create } from "react-test-renderer";
const tree = create(
  <CartItemPreviewComponent name="Name" price="price" quantity="3" _id="id" />
);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
