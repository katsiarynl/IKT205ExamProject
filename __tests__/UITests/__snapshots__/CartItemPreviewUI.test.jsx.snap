// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 1.6,
          "paddingHorizontal": 6,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          {
            "alignItems": "center",
            "fontSize": 17,
            "fontWeight": "bold",
            "justifyContent": "center",
            "textAlign": "left",
          }
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#F2C94C",
            "borderRadius": 150,
            "padding": 7,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          price
           NOK
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#B71C1C",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        -
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 30,
          "marginHorizontal": "5%",
        }
      }
    >
      3
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#B71C1C",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
